csrc/ops/arith.cpp

```diff
@@ -1359,10 +1359,17 @@ TensorView* reductionOp(
     }
   }
 
+  if (keep_dim && offset < 0) {
+    // There were squeezed dimension removed from squeeze that will not be
+    // restored by reductionOpRaw above, so we restore them here
+    out = broadcast(out, is_squeeze);
+  }
```

Comment:
Fixes the failure when `keepdim=True`, as seen in integration test failure (now added to frontend test).

Reply:
I don't think there's any particular preference here.